transport/control.go

```diff
@@ -56,43 +56,33 @@ type windowUpdate struct {
 	increment uint32
 }
 
-func (windowUpdate) isItem() bool {
-	return true
-}
+func (*windowUpdate) isItem() {}
 
 type settings struct {
 	ack bool
 	ss  []http2.Setting
 }
 
-func (settings) isItem() bool {
-	return true
-}
+func (*settings) isItem() {}
 
 type resetStream struct {
 	streamID uint32
 	code     http2.ErrCode
 }
 
-func (resetStream) isItem() bool {
-	return true
-}
+func (*resetStream) isItem() {}
 
 type flushIO struct {
 }
```

Comment:
That section of the FAQ is completely irrelevant. This method is never
called, this is about interface-induced allocation and type safety.
On Apr 1, 2016 17:01, "Qi Zhao" <notifications@github.com> wrote:

> In transport/control.go
> <https://github.com/grpc/grpc-go/pull/615#discussion_r58266999>:
>
> >
> >  type resetStream struct {
> >  	streamID uint32
> >  	code     http2.ErrCode
> >  }
> >
> > -func (resetStream) isItem() bool {
> > -	return true
> > -}
> > +func (*resetStream) isItem() {}
> >
> >  type flushIO struct {
> >  }
>
> I am still not convinced we should do this. According to golang faq,
> "For types such as basic types, slices, and small structs, a value
> receiver is very cheap so unless the semantics of the method requires a
> pointer, a value receiver is efficient and clear."
>
> Actually I am thinking we should change the caller's code instead.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc-go/pull/615/files/27c1c446d89d2e3f1e86424b9b02a79e7d8899f6#r58266999>
>